Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -1294,9 +1300,15 @@ void NetworkStorageManager::setStorageSiteValidationEnabledInternal(bool enabled
 {
     assertIsCurrent(workQueue());
 
-    m_storageSiteValidationEnabled = enabled;
-}
+    auto currentEnabled = !!m_allowedSitesForConnections;
+    if (currentEnabled == enabled)
+        return;
 
+    if (enabled)
+        m_allowedSitesForConnections = HashMap<IPC::Connection::UniqueID, HashSet<WebCore::RegistrableDomain>> { };
```

Comment:
Would be good to add an alias for `HashMap<IPC::Connection::UniqueID, HashSet<WebCore::RegistrableDomain>>`